### PR TITLE
Remove all variable-length arrays on the stack

### PR DESCRIPTION
**Pull request by henrycg**
This commit removes all stack-allocated VLAs.

Last week, @sadjad convinced me that stack-allocated arrays are dangerous: If an attacker can control the size of the allocated array, the attacker can allocate a huge array on the stack and crash the program using a stack overflow.

I don't think that this is a terribly pressing concern for libprio, but since MSVC doesn't support VLAs anyways, we might as well get rid of them entirely.

### CODE REVIEW

**Comment by rhelmer** [comment]
You can ignore the failed appveyor build, I haven't finished hooking that up yet in issues 42 (I didn't expect it to start trying heh)

**Comment by rhelmer** [comment]
I'll take a look over the patch tomorrow, I kicked off a Firefox tryserver CI run in the meantime:
https://treeherder.mozilla.org/#/jobs?repo=try&revision=941841840f590e6615bae6e131f114f8f6cbfbe6

Note that this picks up 2a49189144fc8e4ee266e9180de11e045a82c0ac too - since these two changesets passed the Travis-CI tests I am not too worried, but this will at least exercise the Firefox integration code and unit test.

File: pclient/main.c [hunk 1]
```diff
@@ -49,118 +52,116 @@ verify_full(void)
   // Number of clients to simulate.
   const int nclients = 10;
 
-  // New scope to avoid goto weirdness
```
[the remark this answers is not in the document]
Comment (rhelmer) [reply pclient/main.c]: It'd be easier to see the change here as two separate commits in any case :)

File: ptest/client_test.c [hunk 2]
```diff
@@ -80,47 +83,48 @@ test_client_agg(int nclients)
...
-      mu_check(output[i] == v * nclients);
-    }
+  P_CHECKA(output = calloc(ndata, sizeof(unsigned long)));
+  mu_check(PrioTotalShare_final(cfg, output, tA, tB) == SECSuccess);
+  for (int i = 0; i < ndata; i++) {
+    unsigned long v = ((i % 3 == 1) || (i % 5 == 3));
+    mu_check(output[i] == v * nclients);
   }
 
 // rv = SECFailure;
```
Comment (rhelmer) [inline ptest/client_test.c]: Maybe remove these two commented-out lines while you're in here, unless they are helpful for debugging?

File: pclient/main.c [hunk 3]
```diff
@@ -192,6 +192,10 @@ verify_full(void)
     free(for_server_a);
   if (for_server_b)
     free(for_server_b);
+  if (output)
```
[the remark this answers is not in the document]
Comment (henrycg) [reply pclient/main.c]: Yes, enforcing this everywhere sounds good to me.

File: ptest/client_test.c (hunk 2 above)
Comment (henrycg) [reply ptest/client_test.c]: Done.

**Comment by henrycg** [comment]
I just split the commit into two: one to remove the VLAs and one to fix the formatting.

**Pull request merged**